### PR TITLE
RUMM-219 Add activity screen tracker

### PR DESCRIPTION
**Pull request by mariusc83**
### What does this PR do?

This PR adds the Automatic ActivityScreenTracker as an optional feature to the SDK. By enabling this feature we will be able to detect activity changes and start/stop RumView events whenever on the Activity onResume/onPause lifecycle events.

### Review checklist (to be filled by reviewers)

- [x] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [x] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [x] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: dd-sdk-android/src/test/kotlin/com/datadog/android/DatadogConfigBuilderTest.kt [hunk 1]
```diff
@@ -581,4 +583,67 @@ class DatadogConfigBuilderTest {
...
+                DatadogConfig.RumConfig(
+                    fakeClientToken,
+                    fakeApplicationId,
+                    rumUrl,
+                    DatadogConfig.DEFAULT_SERVICE_NAME,
+                    DatadogConfig.DEFAULT_ENV_NAME,
+                    trackFragmentsAsScreens = true
+                )
+            )
+    }
```
Comment (ncreated) [inline dd-sdk-android/src/test/kotlin/com/datadog/android/DatadogConfigBuilderTest.kt]: Don't we miss tests for default values of `trackGestures`, `trackActivitiesAsScreens` and `trackFragmentsAsScreens`?

Comment (mariusc83) [reply dd-sdk-android/src/test/kotlin/com/datadog/android/DatadogConfigBuilderTest.kt]: Good catch ;)

Comment (mariusc83) [reply dd-sdk-android/src/test/kotlin/com/datadog/android/DatadogConfigBuilderTest.kt]: Done ;)

File: dd-sdk-android/src/test/kotlin/com/datadog/android/DatadogConfigBuilderTest.kt [hunk 2]
```diff
@@ -581,4 +583,89 @@ class DatadogConfigBuilderTest {
...
+
+        assertThat(config.rumConfig)
+            .isEqualTo(
+                DatadogConfig.RumConfig(
+                    fakeClientToken,
+                    fakeApplicationId,
+                    rumUrl,
+                    DatadogConfig.DEFAULT_SERVICE_NAME,
+                    DatadogConfig.DEFAULT_ENV_NAME,
+                    trackActivitiesAsScreens = true
```
Comment (xgouchet) [inline dd-sdk-android/src/test/kotlin/com/datadog/android/DatadogConfigBuilderTest.kt]: Do we want to allow tracking both screens and activities as view ? The way the RUM monitor works, only one view can be active at any time, meaning that you'll start a view for an activity, and immediately after start a view for a fragment.

File: dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/instrumentation/TrackingStrategiesTest.kt [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    internal class Test3Activity() : Activity()
+}
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(Configurator::class)
+internal class ActivityTrackingStrategyTest : TrackingStrategiesTest() {
```
Comment (xgouchet) [inline dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/instrumentation/TrackingStrategiesTest.kt]: this one should be in its own file to make things more readable

File: dd-sdk-android/src/test/kotlin/com/datadog/android/DatadogConfigBuilderTest.kt (hunk 2 above)
Comment (mariusc83) [reply dd-sdk-android/src/test/kotlin/com/datadog/android/DatadogConfigBuilderTest.kt]: That's a good point here, I don't think we should allow both in the same time. I will change in the way that if you selected "trackFragmentsAsScreens" to override the "trackActivityAsScreens" flag to false (meaning that fragments screens have priority). I will state this in the method docs also.

File: dd-sdk-android/src/main/kotlin/com/datadog/android/DatadogConfig.kt [hunk 4]
```diff
@@ -238,13 +248,55 @@ private constructor(
...
             if (endpoint.startsWith("http://")) {
                 needsClearTextHttp = true
             }
         }
     }
 
+    /**
+     * The available strategies for the automatic View tracker.
+     */
+    enum class ViewTrackerStrategy {
```
Comment (xgouchet) [inline dd-sdk-android/src/main/kotlin/com/datadog/android/DatadogConfig.kt]: This is total nit-picking, and you can change that in a future PR if you prefer, but I'd name that `ViewTrackingStrategy`. It is a strategy on how to track views, so it should come with a verb.

Comment (mariusc83) [reply dd-sdk-android/src/main/kotlin/com/datadog/android/DatadogConfig.kt]: Will change these names in my next PR

**Pull request merged**